internal/compiler/program.go

```diff
@@ -161,7 +161,9 @@ func NewProgram(options ProgramOptions) *Program {
 	for _, file := range p.files {
 		extension := tspath.TryGetExtensionFromPath(file.FileName())
 		if extension == tspath.ExtensionTsx || slices.Contains(tspath.SupportedJSExtensionsFlat, extension) {
-			p.unsupportedExtensions = append(p.unsupportedExtensions, extension)
+			if !slices.Contains(p.unsupportedExtensions, extension) {
+				p.unsupportedExtensions = append(p.unsupportedExtensions, extension)
+			}
```

Comment:
Wow, this is my bad. Just do `core.AppendIfUnique` here.

Reply:
I was making the change but realized you did it already :)

Thanks!